crates/chia-secp/src/secp256k1/public_key.rs

```diff
@@ -0,0 +1,59 @@
+use std::fmt;
+use std::hash::{Hash, Hasher};
+
+use chia_sha2::Sha256;
+use k256::ecdsa::signature::hazmat::PrehashVerifier;
+use k256::ecdsa::{Error, VerifyingKey};
+
+use super::Secp256k1Signature;
+
+#[derive(Clone, Copy, PartialEq, Eq)]
+pub struct Secp256k1PublicKey(pub(crate) VerifyingKey);
+
+impl Hash for Secp256k1PublicKey {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.to_bytes().hash(state);
+    }
+}
+
+impl fmt::Debug for Secp256k1PublicKey {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "Secp256k1PublicKey({self})")
+    }
+}
+
+impl fmt::Display for Secp256k1PublicKey {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", hex::encode(self.to_bytes()))
+    }
+}
+
```

Comment:
I think it would be easy to have tests for these

Reply:
Added the tests for these